src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -923,13 +921,16 @@ public ProjectState UpdateDocumentsOrder(ImmutableList<DocumentId> documentIds)
         TextDocumentStates<AdditionalDocumentState> newAdditionalDocumentStates,
         ImmutableArray<TextDocumentState> oldDocuments,
         ImmutableArray<TextDocumentState> newDocuments,
-        bool contentChanged,
         out AsyncLazy<VersionStamp> dependentDocumentVersion,
         out AsyncLazy<VersionStamp> dependentSemanticVersion)
     {
         var recalculateDocumentVersion = false;
         var recalculateSemanticVersion = false;
 
+        var contentChanged = !oldDocuments.SequenceEqual(
+            newDocuments,
+            static (oldDocument, newDocument) => oldDocument.TextAndVersionSource == newDocument.TextAndVersionSource);
```

Comment:
>oldDocument.TextAndVersionSource == newDocument.TextAndVersionSource [](http://example.com/codeflow?start=49&length=68)

This seems a bit odd. @CyrusNajmabadi -- is this an ok comparison?

Reply:
don't know :)  doens't seem like a terrible approach though.   if they're thes ame, they have the same content.  if they're differernt, they're likely to be different.

Reply:
OK, then this looks good. I wasn't sure about the "likely to be different" part of your statement